### PR TITLE
requirements: bump craft-cli to fix logging in overlay

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: tests/spread/general/overlay-logs/task.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+summary: Check that actions done inside the overlay are logged
+
+execute: |
+  error_pattern="E: Unable to locate package idontexist"
+  rockcraft --verbosity=debug 2>&1 >/dev/null | MATCH "$error_pattern"
+  rockcraft_log_file=$(find /root/.local/state/rockcraft/log/ -name 'rockcraft*.log' | sort -n | tail -n1)
+  cat "$rockcraft_log_file" | MATCH "$error_pattern"
```
Comment (lengau) [inline tests/spread/general/overlay-logs/task.yaml:7]: Hi there, your local neighbourhood busybody here to let you know that `MATCH` actually runs `grep -q -E` [under the hood](https://github.com/snapcore/spread/blob/ded9133cdbceaf01f8a1c9decf6ff9ea56e194d6/spread/client.go#L790).

```suggestion
  MATCH "$error_pattern" "$rockcraft_log_file"
```

I don't have anything valuable to add. Just saw the lint failure while I was trying to find an unrelated build failure

Comment (tigarmo) [reply tests/spread/general/overlay-logs/task.yaml]: oh of course! I have a blind spot with `MATCH` sometimes

File: requirements-dev.txt [hunk 2]
```diff
@@ -17,7 +17,8 @@ colorama==0.4.6
 commonmark==0.9.1
 coverage==6.5.0
 craft-archives==1.1.0
-craft-cli==2.0.0
+#craft-cli==2.0.0
```
Comment (tigarmo) [inline requirements-dev.txt:20]: reminding myself to update this

**Review by lengau — APPROVED** [review]
This is great! I also really love the fact that you added the extra spread test here.

That has me thinking about ways to catch issues with our apps in CI...

**Review by mr-cal — APPROVED** [review]

**Pull request merged**